test/taskrun_test.go

```diff
@@ -74,35 +74,35 @@ func TestTaskRunFailure(t *testing.T) {
 	expectedStepState := []v1alpha1.StepState{{
 		ContainerState: corev1.ContainerState{
 			Terminated: &corev1.ContainerStateTerminated{
-				ExitCode: 1,
-				Reason:   "Error",
+				ExitCode: 0,
+				Reason:   "Completed",
 			},
 		},
-		Name: "exit",
+		Name: "nop",
```

Comment:
Just wondering if step order should change?
Now as per this fix `nop` step is appearing at the top. If the user fetches the logs it would show the `nop` step first?
Is it desired? :thinking:

Reply:
Good question. I feel it's no big deal (as the order was always alphabitically and not from the step order), but… @bobcatfish :pray:

Reply:
it doesn't seem like we have any control over this, so I don't think it matters (does anyone understand why these are in the order they're in? it seems pretty random!)

Reply:
yeah its because of task reconciler copies the `pod.status.containerStatus` for all steps. By default, k8s sort the `containerStatus` in alphabetically sorted order (which is wired don't know why).

While working on fetching the task logs, one has to either iterate through `pod.status.containerStatus` or `TaskRun.Status.Steps`. Now the issue is, steps order is not maintained anymore, it has to find the currently running container and rather simply iterate over `taskrun.status.steps` while printing the logs. isn't it wired?  :thinking: 

> it doesn't seem like we have any control over this

One approach is while updating the taskrun status, we can sort containers status as per `task.spec.steps` order.  :thinking: 

WDYT? @vdemeester @bobcatfish

Reply:
hm I guess we could sort them - I expect we'd probably want to have the order of the statuses match the order that the steps are declared in?

anyway my feeling is that we do this in a separate issue and PR, if there are no objections! (I wouldn't think of the order here as part of the interface)

Reply:
Created an issue to follow-up #942